src/test/kotlin/org/rust/ide/annotator/RsErrorAnnotatorTest.kt

```diff
@@ -4278,6 +4280,28 @@ class RsErrorAnnotatorTest : RsAnnotatorTestBase(RsErrorAnnotator::class) {
         }
     """)
 
+    @ProjectDescriptor(WithDependencyRustProjectDescriptor::class)
+    fun `test user defined derive proc macro`() = checkByFileTree("""
+    //- dep-proc-macro/lib.rs
+        use proc_macro::TokenStream;
+
+        #[proc_macro_attribute]
+        pub fn derive(attr: TokenStream, item: TokenStream) -> TokenStream {
+            item
+        }
+    //- lib.rs
+        #[dep_proc_macro::derive]
+        fn main() {
+        }
+
+        mod foo {
+            use dep_proc_macro::derive;
+
+            #[derive/*caret*/]
```

Comment:
```suggestion
            #[derive]
```

Reply:
`checkByFileTree` requires caret marker, or did I put it in the wrong place?

Reply:
Oh, indeed. Lets put it in the beginning of the file
```
    //- lib.rs
        /*caret*/
        #[dep_proc_macro::derive]
        fn main() {}
        ...
```